src/network/bmproto.py

```diff
@@ -460,17 +461,17 @@ def peerValidityChecks(self):
                 errorText="Your time is too far in the future compared to mine. Closing connection."))
             logger.info("%s's time is too far in the future (%s seconds). Closing connection to it.",
                 self.destination, self.timeOffset)
-            shared.timeOffsetWrongCount += 1
+            self.timeOffsetWrongCount += 1
```

Comment:
This breaks things because it's now per connection, whereas it should be global. I suggest you use a class variable, i.e.
`BMProto.timeOffsetWrongCount` instead of `self.timeOffsetWRongCount`. Based on googling (although I haven't tested it) it should work. Besides, I'm not sure I fully tested this variable after moving to asyncore.

Reply:
I didn't found where this variable used at all.

Reply:
Yes setting `BMProto.timeOffsetWrongCount` works as global variable.
```
In [31]: class TestVar(object):
    ...:     var = 0
    ...:     def inc(self):
    ...:         TestVar.var += 1
    ...:         

In [32]: t1 = TestVar()

In [33]: t2 = TestVar()

In [34]: t1.var
Out[34]: 0

In [35]: t1.inc()

In [36]: t1.var
Out[36]: 1

In [37]: t2.var
Out[37]: 1
```

Reply:
So I probably forgot finishing implementing this. Just leave it as a class variable then and I'll open an issue about finishing this.